pkg/agent/util/net_windows_test.go

```diff
@@ -0,0 +1,1129 @@
+		},
+		{
+			name:                "Adapter Err",
+			nodeIPNet:           &ipv4PublicIPNet,
+			dnsServers:          testDNSServer,
+			ipFound:             true,
+			testNetInterfaceErr: testInvalidErr,
+			wantErr:             testInvalidErr,
+		},
+		{
+			name:       "Rename Err",
+			nodeIPNet:  &ipv4PublicIPNet,
+			dnsServers: testDNSServer,
+			newName:    "test1",
+			ipFound:    true,
+			commandErr: testInvalidErr,
+			wantErr:    testInvalidErr,
+		},
+		{
+			name:                   "Enable HNS Err",
+			nodeIPNet:              &ipv4PublicIPNet,
+			dnsServers:             testDNSServer,
+			ipFound:                true,
+			hnsNetworkRequestError: testInvalidErr,
+			wantErr:                testInvalidErr,
+		},
+		{
+			name:       "Enable RSC Err",
+			nodeIPNet:  &ipv4PublicIPNet,
+			dnsServers: testDNSServer,
+			ipFound:    true,
+			commandErr: testInvalidErr,
+			wantErr:    testInvalidErr,
+		},
+		{
+			name:       "IP Not Found",
+			nodeIPNet:  &ipv4ZeroIPNet,
+			dnsServers: testDNSServer,
+			ipFound:    false,
+			wantErr:    nil,
+		},
+		{
+			name:       "IP Not Found Configure Default Err",
+			nodeIPNet:  &ipv4ZeroIPNet,
+			dnsServers: testDNSServer,
+			ipFound:    false,
+			commandErr: testInvalidErr,
+			wantErr:    testInvalidErr,
+		},
+		{
+			name:       "IP Not Found Set Adapter Err",
+			nodeIPNet:  &ipv4ZeroIPNet,
+			dnsServers: testDNSServer,
+			ipFound:    false,
+			commandErr: alreadyExistsErr,
+			wantErr:    alreadyExistsErr,
+		},
+		{
+			name:       "IP Not Found New Net Route Err",
+			nodeIPNet:  &ipv4ZeroIPNet,
+			ipFound:    false,
+			commandErr: alreadyExistsErr,
+			wantErr:    alreadyExistsErr,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runCommand = func(cmd string) (string, error) {
+				return "success", tc.commandErr
+			}
+			netInterface = func() ([]net.Interface, error) {
+				testNetInterfaces, mockNetInterfaces := generateNetInterfaces()
+				for idx, iface := range testNetInterfaces {
+					testNetInterfaces[idx].Name = "vEthernet" + iface.Name
+					mockNetInterfaces[idx].Name = testNetInterfaces[idx].Name
+				}
+				tc.testNetInterfaces = testNetInterfaces
+				return mockNetInterfaces, tc.testNetInterfaceErr
+			}
+			netInterfaceAddrs = func(i *net.Interface) ([]net.Addr, error) {
+				if tc.ipFound {
+					for _, itf := range tc.testNetInterfaces {
+						if itf.Name == i.Name {
+							netAddrs, _ := itf.Addrs()
+							return netAddrs, nil
+						}
+					}
+				}
+				return []net.Addr{}, nil
+			}
+			hnsNetworkRequest = func(method, path, request string) (*hcsshim.HNSNetwork, error) {
+				return nil, tc.hnsNetworkRequestError
+			}
+			hnsNetworkCreate = func(network *hcsshim.HNSNetwork) (*hcsshim.HNSNetwork, error) {
+				return network, tc.hnsNetworkCreateErr
+			}
+			hnsNetworkDelete = func(network *hcsshim.HNSNetwork) (*hcsshim.HNSNetwork, error) {
+				return network, nil
+			}
+			defer func() {
```

Comment:
Please follow the suggested mock function format in Windows, too.

Reply:
Refactored